Doc/tools/extensions/implementation_detail.py

```diff
@@ -0,0 +1,47 @@
+"""Support for marking up implementation details."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from docutils import nodes
+from sphinx.locale import _ as sphinx_gettext
+from sphinx.util.docutils import SphinxDirective
+
+if TYPE_CHECKING:
```

Comment:
Maybe avoid the `typing` import?
```suggestion
from docutils import nodes
from sphinx.locale import _ as sphinx_gettext
from sphinx.util.docutils import SphinxDirective

TYPE_CHECKING = False
if TYPE_CHECKING:
```

Reply:
I'd prefer to do this for every extension at once, but no objection in principle. I'll add this to the list of clean-ups for the end of this project.